### PR TITLE
Cleanup api

### PR DESCRIPTION
**Pull request by mdellweg**
While thinking about the `ForemanStatelessEntityAnsibleModule` i found some cruft to cleanup...

### CODE REVIEW

**Comment by mdellweg** [comment]
CI failure unrelated

**Comment by evgeni** [comment]
the `__init__.py` drop breaks tests on python 2.7:

```
______________ ERROR collecting tests/test_foreman_spec_helper.py ______________
ImportError while importing test module '/home/runner/work/foreman-ansible-modules/foreman-ansible-modules/tests/test_foreman_spec_helper.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
tests/test_foreman_spec_helper.py:1: in <module>
    from plugins.module_utils.foreman_helper import _foreman_spec_helper
E   ImportError: No module named plugins.module_utils.foreman_helper
===================== 106 passed, 1 error in 2.77 seconds ======================
```

Ansible itself is fine with that tho.

**Comment by mdellweg** [comment]
Oh right. That is where we still test the original files.

File: plugins/module_utils/foreman_helper.py [hunk 1]
```diff
@@ -971,10 +962,11 @@ def run(self, **kwargs):
                 self.foreman_params[self.entity_key] = self.foreman_params.pop(updated_key)
 
         params = kwargs.get('params', {})
-        if self.entity_scope:
-            for scope in self.entity_scope:
+        entity_scope = self.foreman_spec['entity'].get('scope')
+        if entity_scope:
+            for scope in entity_scope:
```
Comment (evgeni) [inline plugins/module_utils/foreman_helper.py:967]: you could write `for scope in self.foreman_spec['entity'].get('scope', []):` here… but I don't think it's needed

Comment (mdellweg) [reply plugins/module_utils/foreman_helper.py:967]: Do you think, it's more readable? Because that's the only reason why i would do it.

Comment (evgeni) [reply plugins/module_utils/foreman_helper.py:967]: It saves two lines, whether this is more readable depends on the reader, I guess. So probably just leave it as it is.

**Pull request merged**